kernel/kernel-api/src/main/java/io/delta/kernel/internal/DeltaErrors.java

```diff
@@ -293,6 +293,12 @@ public static ConcurrentWriteException concurrentDomainMetadataAction(
     return new ConcurrentWriteException(message);
   }
 
+  public static KernelException rowIDAssignmentWithoutStats() {
+    return new KernelException(
+        "All AddFile actions must have statistics that include the number of records "
```

Comment:
This is unfortunately not specified in Protocol, but it would be difficult to assign fresh row IDs without knowing `numRecords` (Delta-Spark also errors out in this case), so this could be an issue.

There is actually a question about this raised [here](https://github.com/delta-io/delta/pull/3894#discussion_r1852593095). To summarize, 
1. do we have some guarantee or requirement that connectors populate `numRecords`? Are connectors that implement writes today (if any) populating `numRecords`?
2. can we puts the burden of populating `numRecords` more on the connector?

Curious to know what do you think, thanks!